core/test-api/karaf/src/main/java/org/opennms/core/test/karaf/KarafTestCase.java

```diff
@@ -249,10 +259,27 @@ private static int getAvailablePort(int min, int max) {
             editConfigurationFilePut("etc/org.apache.karaf.management.cfg", "rmiServerPort", rmiServerPort),
             editConfigurationFilePut("etc/org.apache.karaf.shell.cfg", "sshPort", sshPort),
 
+            // Work around bug KARAF-5251
+            editConfigurationFilePut("etc/startup.properties", "mvn:net.java.dev.jna/jna/4.4.0", "5"),
+            editConfigurationFilePut("etc/startup.properties", "mvn:net.java.dev.jna/jna-platform/4.4.0", "5"),
+
             // This port is already being allocated according to an org.ops4j.net.FreePort call
             //editConfigurationFilePut("etc/system.properties", "org.ops4j.pax.exam.rbc.rmi.port", paxExamRmiRegistryPort),
+
+            // Work around bug KARAF-5223, should be unnecessary once we upgrade past Karaf 4.1.1
+            replaceConfigurationFile("etc/shell.init.script", emptyFile),
         };
 
+        // Work around bug KARAF-5384
+        // If there is a MINA JAR available, then add MINA to the startup classpath
+        // TODO: Don't hardcode the version number here
+        File minaJar = new File("target/dependency/mina-core-2.0.16.jar");
```

Comment:
What happens if we upgrade Mina and forget to update this block? I'm guessing that the Karaf integration tests will run unreliably.

Reply:
They should fail 100% without that.